vsintegration/src/FSharp.Editor/LanguageService/LanguageService.fs

```diff
@@ -400,24 +494,13 @@ and
     override this.DebuggerLanguageId = DebuggerEnvironment.GetLanguageID()
 
     override this.CreateContext(_,_,_,_,_) = raise(System.NotImplementedException())
-
-    override this.SetupNewTextView(textView) =
-        base.SetupNewTextView(textView)
-
-        let textViewAdapter = package.ComponentModel.GetService<IVsEditorAdaptersFactoryService>()
-               
-        match textView.GetBuffer() with
-        | (VSConstants.S_OK, textLines) ->
-            let filename = VsTextLines.GetFilename textLines
-            match VsRunningDocumentTable.FindDocumentWithoutLocking(package.RunningDocumentTable,filename) with
-            | Some (hier, _) ->
-                match hier with
-                | :? IProvideProjectSite as siteProvider when not (IsScript(filename)) -> 
-                    this.SetupProjectFile(siteProvider, this.Workspace)
-                | _ -> 
-                    let fileContents = VsTextLines.GetFileContents(textLines, textViewAdapter)
-                    this.SetupStandAloneFile(filename, fileContents, this.Workspace, hier)
-            | _ -> ()
+    
+    member this.TrackDocument(fileName) =
+        match VsRunningDocumentTable.FindDocumentWithoutLocking(package.RunningDocumentTable, fileName) with
```

Comment:
@dsyme this file essentially encapsulates the correctness fixes. However with 15.3 now in preview we can move over to the Miscellaneous Files workspace. Read the comments on https://github.com/dotnet/roslyn/pull/17263 and my comment about Miscellaneous Files in this file.